### PR TITLE
add yarn install method

### PR DESCRIPTION
**Pull request by Borjagodoy**
New method yarnInstall  allow the posibility of use npm or yarn package manage.

The user can chose the package manage across the option Object. If want use yarn and bower install in the method installDependences,  have set the option object with `yarn: true` and `bower: true`.

The option object have is default `{npm:true, bower: true}`

### CODE REVIEW

File: lib/actions/install.js [hunk 1]
```diff
@@ -86,6 +86,8 @@ install.runInstall = function (installer, paths, options, cb) {
  */
 
 install.installDependencies = function (options) {
+  console.log('=========================');
```
Comment (hemanth) [inline lib/actions/install.js]: Needs cleanup.

File: lib/actions/install.js [hunk 2]
```diff
@@ -101,12 +101,14 @@ install.installDependencies = function (options) {
     };
   }
 
-  options = _.defaults(options || {}, {
-    bower: true,
-    npm: true,
-    skipMessage: false,
-    callback: function () {}
-  });
+  if (!options) {
```
Comment (SBoudrias) [inline lib/actions/install.js]: That's not required

Comment (SBoudrias) [inline lib/actions/install.js]: Maybe the right approach then is to not set default values for these and check if they're specifically set as false or true.

Comment (Borjagodoy) [inline lib/actions/install.js]: Ok I understand this line yet. In the last commit I delete the npm: true by default.  The user will have to set npm: or yarn: true depends to the manager that want to use. Other posible solutions is leave option as it is and when the user want use yarn, put yarn: true and npm: false. who is the better option?

File: lib/actions/install.js [hunk 3]
```diff
@@ -103,26 +103,34 @@ install.installDependencies = function (options) {
...
   options = _.defaults(options || {}, {
     bower: true,
-    npm: true,
+    npm: false,
     skipMessage: false,
     callback: function () {}
   });
 
+
   if (options.npm) {
```
Comment (Borjagodoy) [inline lib/actions/install.js]: Yes, I understand you, thanks 👍 

My question now, Is correct the new code? I'm not sure if is the best practise, set for default npm to false.

**Comment by SBoudrias** [comment]
Hey, I see you added commits and haven't address my comments. Just want to make sure you saw them - thanks!

**Comment by mischah** [comment]
Ping 🔔

😘

**Comment by Borjagodoy** [comment]
@SBoudrias I see the comment but i don understand wel. Yo said that is better don't init  the option object and test if there object is not null in  `option == bower'  ' option == bower'  ' option == bower' ?

**Comment by SBoudrias** [comment]
What I mean is delete the `_.defaults`. And instead check if a boolean was explicitly set `if (options.bower === false)` (and same for all other values).

File: lib/actions/install.js [hunk 4]
```diff
@@ -101,28 +101,28 @@ install.installDependencies = function (options) {
...
 
-  options = _.defaults(options || {}, {
-    bower: true,
-    npm: true,
-    skipMessage: false,
-    callback: function () {}
-  });
-
-  if (options.npm) {
+  if (options.npm === true) {
```
Comment (SBoudrias) [inline lib/actions/install.js]: This needs to be `!== false` so the behavior stays backward compatible.

File: lib/actions/install.js [hunk 5]
```diff
@@ -101,28 +101,28 @@ install.installDependencies = function (options) {
...
 
-  if (options.bower) {
+  if (options.yarn === true) {
+    msg.commands.push('yarn install');
+    commands.push(function (cb) {
+      this.yarnInstall(null, null, cb);
+    }.bind(this));
+  }
+
+  if (options.bower === true) {
```
Comment (SBoudrias) [inline lib/actions/install.js]: This needs to be `!== false` so the behavior stays backward compatible.

File: test/install.js [hunk 6]
```diff
@@ -135,25 +152,55 @@ describe('generators.Base (actions/install mixin)', function () {
       this.dummy.run();
     });
   });
+  describe('#yarnInstall()', function () {
```
Comment (SBoudrias) [inline test/install.js]: Please add a space between `describe` blocks.

**Comment by SBoudrias** [comment]
Manually merge with some further code fixes here https://github.com/yeoman/generator/commit/4eccb2a81b4fdb652ebf618a3dd7be7aa7950a96

**Comment by mischah** [comment]
🎉

**Comment by abdonrd** [comment]
🎉

**Pull request closed without merging**